File: sdk/python/examples/quickstart.py
"""Quickstart for the Scratchpad Python SDK.

Requires a running Scratchpad server on :8080 (cmd/server).

Run with the SDK on the path:

    PYTHONPATH=.. python quickstart.py
"""

import sys

from scratchpad_sdk import ScratchpadClient, ScratchpadError

BASE_URL = "http://localhost:8080"


def main() -> None:
    client = ScratchpadClient(BASE_URL)

    # Health check first.
    print("healthz:", client.healthz())

    # Create a session (Chrome, headful), run it, then close it.
    sid = client.create_session(headless=False)
    print("created session:", sid)
    try:
        client.navigate("https://example.com", session_id=sid)

        # Observe: read the interactive elements from the accessibility tree.
        obs = client.observe(session_id=sid)
        for node in obs.spatial_tree:
            if node.interactive:
                print(f"  {node.node_id}: role={node.role} name={node.name!r}")

        # A coordinate click, then read the console ring buffer.
        client.click(x=320, y=180, session_id=sid)
        console = client.get_console(session_id=sid)
        print("console entries:", len(console.get("logs", [])))
    except ScratchpadError as exc:
        # The typed error envelope surfaces code / message / hint / request_id.
        print("error:", exc.code, exc.message, exc.hint, exc.request_id, file=sys.stderr)
        sys.exit(1)
    finally:
        client.delete_session(session_id=sid)
        print("deleted session:", sid)


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/quickstart.ts
/**
 * Quickstart for the Scratchpad TypeScript SDK.
 *
 * Requires a running Scratchpad server on :8080 (cmd/server).
 *
 *   npm install
 *   npx tsx examples/quickstart.ts      # or: npm run build && node dist/...
 */

import { ScratchpadClient, ScratchpadError } from "../src";

async function main(): Promise<void> {
  const client = new ScratchpadClient("http://localhost:8080");

  // Health check first.
  console.log("healthz:", await client.healthz());

  // Create a session (Chrome, headful), run it, then close it.
  const sid = await client.createSession({ headless: false });
  console.log("created session:", sid);
  try {
    await client.navigate("https://example.com", sid);

    // Observe: read the interactive elements from the accessibility tree.
    const obs = await client.observe(sid);
    for (const node of obs.spatial_tree ?? []) {
      if (node.interactive) {
        console.log(`  ${node.node_id}: role=${node.role} name=${JSON.stringify(node.name)}`);
      }
    }

    // A coordinate click, then read the console ring buffer.
    await client.click(320, 180, {}, sid);
    const consoleResp = await client.getConsole(undefined, sid);
    console.log("console entries:", consoleResp.logs.length);
  } catch (err) {
    // The typed error envelope surfaces code / message / hint / requestId.
    if (err instanceof ScratchpadError) {
      console.error("error:", err.code, err.message, err.hint, err.requestId);
    }
    throw err;
  } finally {
    await client.deleteSession(sid);
    console.log("deleted session:", sid);
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
